src/main/java/org/opensearch/timeseries/model/TimeSeriesTask.java

```diff
@@ -0,0 +1,446 @@
+
+        public T lastUpdateTime(Instant lastUpdateTime) {
+            this.lastUpdateTime = lastUpdateTime;
+            return (T) this;
+        }
+
+        public T startedBy(String startedBy) {
+            this.startedBy = startedBy;
+            return (T) this;
+        }
+
+        public T stoppedBy(String stoppedBy) {
+            this.stoppedBy = stoppedBy;
+            return (T) this;
+        }
+
+        public T error(String error) {
+            this.error = error;
+            return (T) this;
+        }
+
+        public T state(String state) {
+            this.state = state;
+            return (T) this;
+        }
+
+        public T taskProgress(Float taskProgress) {
+            this.taskProgress = taskProgress;
+            return (T) this;
+        }
+
+        public T initProgress(Float initProgress) {
+            this.initProgress = initProgress;
+            return (T) this;
+        }
+
+        public T currentPiece(Instant currentPiece) {
+            this.currentPiece = currentPiece;
+            return (T) this;
+        }
+
+        public T executionStartTime(Instant executionStartTime) {
+            this.executionStartTime = executionStartTime;
+            return (T) this;
+        }
+
+        public T executionEndTime(Instant executionEndTime) {
+            this.executionEndTime = executionEndTime;
+            return (T) this;
+        }
+
+        public T isLatest(Boolean isLatest) {
+            this.isLatest = isLatest;
+            return (T) this;
+        }
+
+        public T taskType(String taskType) {
+            this.taskType = taskType;
+            return (T) this;
+        }
+
+        public T checkpointId(String checkpointId) {
+            this.checkpointId = checkpointId;
+            return (T) this;
+        }
+
+        public T coordinatingNode(String coordinatingNode) {
+            this.coordinatingNode = coordinatingNode;
+            return (T) this;
+        }
+
+        public T workerNode(String workerNode) {
+            this.workerNode = workerNode;
+            return (T) this;
+        }
+
+        public T entity(Entity entity) {
+            this.entity = entity;
+            return (T) this;
+        }
+
+        public T parentTaskId(String parentTaskId) {
+            this.parentTaskId = parentTaskId;
+            return (T) this;
+        }
+
+        public T estimatedMinutesLeft(Integer estimatedMinutesLeft) {
+            this.estimatedMinutesLeft = estimatedMinutesLeft;
+            return (T) this;
+        }
+
+        public T user(User user) {
+            this.user = user;
+            return (T) this;
+        }
+    }
+
+    public boolean isHistoricalTask() {
+        return taskType.startsWith(TimeSeriesTask.HISTORICAL_TASK_PREFIX);
+    }
+
+    /**
+     * Get detector level task id. If a task has no parent task, the task is detector level task.
+     * @return detector level task id
+     */
+    public String getDetectorLevelTaskId() {
```

Comment:
just confirming this is to check between forecaster and detector task?

Reply:
Good catch. Renamed to getConfigLevelTaskId